### PR TITLE
🎉  Source Shopify: implement Oauth2.0 for Airbyte-Cloud

### LINKED ISSUE #7243: Source Shopify: implement oauth2 java-part
## Tell us about the problem you're trying to solve
What are you trying to do, and why is it hard? A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### LINKED ISSUE #8633: Get Shopify connector on Cloud
think the issue is that we need to redirect the user to {shop_id}.shopify.com for consent instead of whatever we do now, this was recently added as part of https://github.com/airbytehq/airbyte/issues/7796. So there will be a java component here as well and a python/connector component to modify the spec to use the new oauth spec

Based on [priorities here](https://docs.google.com/spreadsheets/d/1JvWV7IAG6fgElO7zP3HyDgdPQLCAIqNcgT_JJJjdC7A/edit#gid=0), key targets are Google Sheets and Shopify. Other connectors will be benched for now.

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolves: https://github.com/airbytehq/airbyte/issues/7243

## How
- added java part ShopifyOAuthFlow.class

## 🚨 User Impact 🚨
- User have to update the connector to `0.1.28` version.
- User have to update the connector's schemas, after version update.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter

- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [X] Documentation updated 
    - [X] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -641,7 +641,7 @@
 - name: Shopify
   sourceDefinitionId: 9da77001-af33-4bcd-be46-6252bf9342b9
   dockerRepository: airbyte/source-shopify
-  dockerImageTag: 0.1.27
+  dockerImageTag: 0.1.28
```
Comment (lazebnyi) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:644]: Should we bump the version after publishing?

**Review by lazebnyi — COMMENTED** [review]
LGTM! One minor comment.

**Review by antixar — CHANGES_REQUESTED** [review]
Could you add integration config for checking of backward compatibility?

**Comment by bazarnov** [comment]
> Could you add integration config for checking of backward compatibility?

It's already there, the python changes for `source-shopify` were introduced way before this PR, with version `0.1.19`, the `acceptance-test-config.yml`, where this facts are tested. Please check this PR for more info: https://github.com/airbytehq/airbyte/pull/6951

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml (hunk 1 above)
Comment (bazarnov) [reply airbyte-config/init/src/main/resources/seed/source_definitions.yaml:644]: I see, thanks for the advice.

**Review by antixar — APPROVED** [review]
LGTM
Please publish your changes

**Pull request merged**